client/http/http.go

```diff
@@ -17,14 +18,14 @@ import (
 	json "github.com/nikkolasg/hexjson"
 )
 
-// NewHTTPClient creates a new client pointing to an HTTP endpoint
-func NewHTTPClient(url string, chainHash []byte, client http.RoundTripper) (Client, error) {
-	if client == nil {
-		client = http.DefaultTransport
+// New creates a new client pointing to an HTTP endpoint
+func New(url string, chainHash []byte, transport nhttp.RoundTripper) (client.Client, error) {
+	if transport == nil {
+		transport = nhttp.DefaultTransport
 	}
 	c := &httpClient{
 		root:   url,
-		client: instrumentClient(url, client),
+		client: instrumentClient(url, transport),
 		l:      log.DefaultLogger,
 	}
 	chainInfo, err := c.FetchChainInfo(chainHash)
```

Comment:
in the CLI constructor (https://github.com/drand/drand/pull/511/files#diff-942292593272d13070d9b0ae8f3c410fR95 ) as soon as we find a client that succeeds in learning info, that info is propagated to others, so the behavior when accessed via the CLI uses (client, relay, relay-gossip) should be the same as before.

i believe the filecoin usage expects to already have group info to provide.

That said - you're right, this is offloading the problem and making the consumer handle it rather than doing it ourselves. I'll add the same logic to the `ForURLs` convenience wrapper to mitigate